src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -526,6 +526,162 @@ private static int GetFloatingPointMaxDigitsAndPrecision(char fmt, ref int preci
             return maxDigits;
         }
 
+        private static void FormatFloatAsHex<TNumber, TChar>(ref ValueListBuilder<TChar> vlb, TNumber value, char fmt, int precision)
+            where TNumber : unmanaged, IBinaryFloatParseAndFormatInfo<TNumber>
+            where TChar : unmanaged, IUtfChar<TChar>
+        {
+            // Get the raw bits
+            ulong bits = TNumber.FloatToBits(value);
+            int mantissaBits = TNumber.NormalMantissaBits;
+            int exponentBias = TNumber.ExponentBias;
+
+            // Extract sign, exponent, and mantissa
+            bool isNegative = (bits >> (mantissaBits + TNumber.ExponentBits)) != 0;
+            int biasedExponent = (int)((bits >> mantissaBits) & ((1UL << TNumber.ExponentBits) - 1));
+            ulong mantissa = bits & TNumber.NormalMantissaMask;
+
+            // Add sign
+            if (isNegative)
+            {
+                vlb.Append(TChar.CastFrom('-'));
+            }
+
+            // Add "0x" prefix
+            vlb.Append(TChar.CastFrom('0'));
+            vlb.Append(TChar.CastFrom(fmt)); // 'x' or 'X'
```

Comment:
We do not emit the `0x` prefix for integers or other number types. We similarly do not emit a `0b` prefix for binary or for any other specialized formats.

As such, we should likely not do this for `HexFloat` either and should instead leave it off.